### ISSUE TITLE
Aliased for tables, attributes and add a short description

### ISSUE THREAD

**Issue by marcel1334** [OHDSI/WhiteRabbit#168]
Sometimes the names of tables and attributes are selfexplaining. But we also see very short names and/or in a local language. It would be handy if the user can add aliases for the table and attribute names so that these can be included in the scan report. Either after the scanning or add this info in a meta datafiles. Also a very short description on table/attribute level would help interpreting the scan report.

**Comment by PRijnbeek** [comment]
Yes, I would support this enhancement. We have encountered this many times when mapping non English databases. We solved this at that time by asking them to provide a data dictionary with their White Rabbit output. It would be nice if White Rabbit would generate a template they can fill in per table/field that RaH can read (JSON?). For example:

- English table name
- Short description of its content

For each column:
- English name
- Short explanation if needed
- Code system if relevant.

etc..

**Comment by MaximMoinat** [comment]
Yes, we should be able to ingest the data dictionary. We provide the format for this, which can be a simple excel file that the data owner can extract from their own dictionary.

**Comment by MaximMoinat** [comment]
After some discussion internally, we have come up with the following approach. Based on the following:
 * The descriptions go into new scan report columns with  an additional overview sheet for tables. Both the field and table level sheet get an alias and description columns.
 * These fields can be populated in three ways:
    * manually, after scan report generation
    * semi-auto, by supplying a data dictionary in fixed format to the WR scan documents
    * auto, by reading comment fields from a database (retrieval might be very rdbms specific)
* RiaH ingests these new columns and shows them similar to how is done for target fields when clicked on. If alias is provided, this can be used instead of the original table/field name.

@marcel1334 Any thoughts? 
Including @spayralbe and @anne0507, as they will be contributing as well.

**Comment by marcel1334** [comment]
Good to hear that aliases and descriptions are on the table!!!

As I understand correctly, the three ways you suggest are done in a separate file/database and must be imported in the scan report first and then that updated scan-report has to be loaded in HiaH before its shown in RiaH. Is that correct?

Technically this will work, but I was thinking about the workflow. I think the aliases and descriptions are not a static thing you only specify before starting RiaH and never touch again. Especially the descriptions will be modified during the ETL design. During the ETL design discussions with the source-db-experts will result in extra info about the source tables and columns you want to add to the table/column descriptions and maybe will even modify the given alias as it turns out a better alias is available. So, I think it would be very handy to edit these source aliases/descriptions from within RiaH (as well), since this is probably the tool on the screen at that moment. 

In this situation, you can change the alias/description and directly continue with the ETL design and discussion with the source-db-expert. Instaed of swithching tools and reimport stuff. At the background I can imaging you can store it in the same scan report as you suggested of course.

Does this make sense and this feasible?

**Comment by MaximMoinat** [comment]
RiaH already provides that functionality, you can add a 'comment' to the source tables and fields. The suggested approach is just a way to bulk-import the available source descriptions. In my view, these descriptions are static (provided one time by the source). 

About the aliases; basically this would be to modifiable table/field names. e.g. when you right-click on a box, you get the option to rename it. I am a bit hesitant about this, as it can give problems with the output later on (especially the test framework).
